src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -637,7 +694,11 @@ private HttpComponentsClientHttpRequestFactory getRequestFactory(final URL url,
       bldr.setSslcontext(sslContext);
     }
 
-    HttpClient httpClient = bldr.build();
+    HttpClient httpClient;
+    // this lets the user perform non-essential configuration (e.g. timeouts)
+    // but reduces the risk of essentials not being set. MK.
+    HttpClientBuilder b = configurator.configure(bldr);
```

Comment:
To be honest that `NoOpHttpClientBuilderConfigurator` could be replaces with just a `if (configurator != null)`